### PR TITLE
Paraswap: Implement custom claimer task

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

File: packages/users/paraswap/fee-collector/contracts/ParaswapClaimer.sol [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * @dev Tells the address from where the token amounts to execute this task are fetched
+     */
+    function getTokensSource() external view virtual override(IBaseTask, BaseTask) returns (address) {
+        return feeClaimer;
+    }
+
+    /**
+     * @dev Tells the balance of the depositor for a given token
```
Comment (lgalende) [inline packages/users/paraswap/fee-collector/contracts/ParaswapClaimer.sol:66]:
```suggestion
     * @dev Tells the balance of the Paraswap claimer for a given token
```

**Review by dmf7z — APPROVED** [review]

File: packages/users/paraswap/fee-collector/contracts/interfaces/IFeeClaimer.sol [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity >=0.8.0;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
```
Comment (lgalende) [inline packages/users/paraswap/fee-collector/contracts/interfaces/IFeeClaimer.sol:17]: Is this `import` necessary?

File: packages/users/paraswap/fee-collector/contracts/interfaces/IParaswapClaimer.sol [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity >=0.8.0;
+
+import '@mimic-fi/v3-tasks/contracts/interfaces/ITask.sol';
```
Comment (lgalende) [inline packages/users/paraswap/fee-collector/contracts/interfaces/IParaswapClaimer.sol:17]: Is this other `import` necessary?

File: packages/users/paraswap/fee-collector/contracts/test/FeeClaimerMock.sol [hunk 4]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/utils/Address.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol';
+
+import '@mimic-fi/v3-helpers/contracts/utils/Denominations.sol';
```
Comment (lgalende) [inline packages/users/paraswap/fee-collector/contracts/test/FeeClaimerMock.sol:9]: Are these ones necessary?

File: packages/users/paraswap/fee-collector/test/ParaswapClaimer.test.ts [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+  describe('setFeeClaimer', () => {
+    context('when the sender is authorized', () => {
+      beforeEach('set sender', async () => {
+        const setFeeClaimerRole = task.interface.getSighash('setFeeClaimer')
+        await authorizer.connect(owner).authorize(owner.address, task.address, setFeeClaimerRole, [])
+        task = task.connect(owner)
+      })
+
+      context('when the given address is not zero', () => {
+        it('sets the protocol fee withdrawer', async () => {
```
Comment (lgalende) [inline packages/users/paraswap/fee-collector/test/ParaswapClaimer.test.ts:56]:
```suggestion
        it('sets the fee claimer', async () => {
```

File: packages/users/paraswap/fee-collector/test/ParaswapClaimer.test.ts [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+              const previousSmartVaultBalance = await token.balanceOf(smartVault.address)
+              const previousFeeWithdrawerBalance = await token.balanceOf(feeClaimer.address)
+
+              await task.connect(owner).call(token.address, requestedAmount)
+
+              const currentSmartVaultBalance = await token.balanceOf(smartVault.address)
+              expect(currentSmartVaultBalance).to.be.eq(previousSmartVaultBalance.add(transactedAmount))
+
+              const currentFeeWithdrawerBalance = await token.balanceOf(feeClaimer.address)
+              expect(currentFeeWithdrawerBalance).to.be.eq(previousFeeWithdrawerBalance.sub(transactedAmount))
```
Comment (lgalende) [inline packages/users/paraswap/fee-collector/test/ParaswapClaimer.test.ts:195]:
```suggestion
            it('transfers the token in from the fee claimer to the smart vault', async () => {
              const previousSmartVaultBalance = await token.balanceOf(smartVault.address)
              const previousFeeClaimerBalance = await token.balanceOf(feeClaimer.address)

              await task.connect(owner).call(token.address, requestedAmount)

              const currentSmartVaultBalance = await token.balanceOf(smartVault.address)
              expect(currentSmartVaultBalance).to.be.eq(previousSmartVaultBalance.add(transactedAmount))

              const currentFeeClaimerBalance = await token.balanceOf(feeClaimer.address)
              expect(currentFeeClaimerBalance).to.be.eq(previousFeeClaimerBalance.sub(transactedAmount))
```

**Review by lgalende — APPROVED** [review]

**Pull request merged**